src/Microsoft.Health.Dicom.Core/Features/Validation/DicomElementMinimumValidator.cs

```diff
@@ -3,67 +3,53 @@
 // -------------------------------------------------------------------------------------------------
 
-using System;
 using System.Collections.Generic;
 using System.Diagnostics;
 using Dicom;
-using Dicom.IO.Buffer;
 using EnsureThat;
-using Microsoft.Health.Dicom.Core.Extensions;
 
 namespace Microsoft.Health.Dicom.Core.Features.Validation
 {
     public class DicomElementMinimumValidator : IDicomElementMinimumValidator
     {
-        private readonly Dictionary<DicomVR, Action<string, string>> _stringValidators = new Dictionary<DicomVR, Action<string, string>>();
-        private readonly Dictionary<DicomVR, Action<IByteBuffer, string>> _byteBufferValidators = new Dictionary<DicomVR, Action<IByteBuffer, string>>();
 
-        public DicomElementMinimumValidator()
+        private static readonly IReadOnlyDictionary<DicomVR, IDicomElementValidation> Validations = new Dictionary<DicomVR, IDicomElementValidation>
         {
-            _stringValidators.Add(DicomVR.AE, DicomElementMinimumValidation.ValidateAE);
-            _stringValidators.Add(DicomVR.AS, DicomElementMinimumValidation.ValidateAS);
-            _stringValidators.Add(DicomVR.CS, DicomElementMinimumValidation.ValidateCS);
-            _stringValidators.Add(DicomVR.DA, DicomElementMinimumValidation.ValidateDA);
-            _stringValidators.Add(DicomVR.DS, DicomElementMinimumValidation.ValidateDS);
-            _stringValidators.Add(DicomVR.IS, DicomElementMinimumValidation.ValidateIS);
-            _stringValidators.Add(DicomVR.LO, DicomElementMinimumValidation.ValidateLO);
-            _stringValidators.Add(DicomVR.PN, DicomElementMinimumValidation.ValidatePN);
-            _stringValidators.Add(DicomVR.SH, DicomElementMinimumValidation.ValidateSH);
-            _stringValidators.Add(DicomVR.UI, DicomElementMinimumValidation.ValidateUI);
-
-            _byteBufferValidators.Add(DicomVR.FL, DicomElementMinimumValidation.ValidateFL);
-            _byteBufferValidators.Add(DicomVR.FD, DicomElementMinimumValidation.ValidateFD);
-            _byteBufferValidators.Add(DicomVR.SL, DicomElementMinimumValidation.ValidateSL);
-            _byteBufferValidators.Add(DicomVR.SS, DicomElementMinimumValidation.ValidateSS);
-            _byteBufferValidators.Add(DicomVR.UL, DicomElementMinimumValidation.ValidateUL);
-            _byteBufferValidators.Add(DicomVR.US, DicomElementMinimumValidation.ValidateUS);
-        }
-
-        // only works for single value dicom element
-        public void Validate(DicomElement element)
+            { DicomVR.AE, new DicomElementMaxLengthValidation(16) },
+            { DicomVR.AS, new DicomElementRequiredLengthValidation(4) },
+            { DicomVR.CS, new DicomElementMaxLengthValidation(16) },
+            { DicomVR.DA, new DicomDateValidation() },
+            { DicomVR.DS, new DicomElementMaxLengthValidation(16) },
+            { DicomVR.FL, new DicomElementRequiredLengthValidation(4) },
```

Comment:
How do we handle more than one type validation for a specific VR? We still need VR specific validation encapsulated in one place for all callers.

Reply:
could you share a case when need need such validation?